provider/docker.go

```diff
@@ -41,13 +42,14 @@ var _ Provider = (*Docker)(nil)
 
 // Docker holds configurations of the Docker provider.
 type Docker struct {
-	BaseProvider     `mapstructure:",squash"`
-	Endpoint         string     `description:"Docker server endpoint. Can be a tcp or a unix socket endpoint"`
-	Domain           string     `description:"Default domain used"`
-	TLS              *ClientTLS `description:"Enable Docker TLS support"`
-	ExposedByDefault bool       `description:"Expose containers by default"`
-	UseBindPortIP    bool       `description:"Use the ip address from the bound port, rather than from the inner network"`
-	SwarmMode        bool       `description:"Use Docker on Swarm Mode"`
+	BaseProvider       `mapstructure:",squash"`
+	Endpoint           string     `description:"Docker server endpoint. Can be a tcp or a unix socket endpoint"`
+	Domain             string     `description:"Default domain used"`
+	TLS                *ClientTLS `description:"Enable Docker TLS support"`
+	ExposedByDefault   bool       `description:"Expose containers by default"`
+	UseBindPortIP      bool       `description:"Use the ip address from the bound port, rather than from the inner network"`
+	SwarmMode          bool       `description:"Use Docker on Swarm Mode"`
+	traefikContainerID string
```

Comment:
I'm not sure if that is a good idea or not. What happens, if you run traefik in cluster mode? How would you handle that case?